plugin/pkg/admission/limitranger/admission.go

```diff
@@ -100,9 +104,67 @@ func (l *LimitRanger) Admit(a admission.Attributes) (err error) {
 		}
 	}
 
+	items, err := l.GetLimitRanges(a)
+	if err != nil {
+		return err
+	}
+
+	// set values for the required limitranges if requested
+	for i := range items {
+		limitRange := items[i]
+
+		if !l.actions.SupportsLimit(limitRange) {
+			continue
+		}
+
+		err = l.actions.MutateLimit(limitRange, a.GetResource().Resource, a.GetObject())
+		if err != nil {
+			return admission.NewForbidden(a, err)
+		}
+	}
+	return nil
+}
+
+// Validate admits resources into cluster that do not violate any defined LimitRange in the namespace
+func (l *LimitRanger) Validate(a admission.Attributes) (err error) {
+	if !l.actions.SupportsAttributes(a) {
+		return nil
+	}
+
+	obj := a.GetObject()
+	name := "Unknown"
+	if obj != nil {
+		name, _ = meta.NewAccessor().Name(obj)
+		if len(name) == 0 {
+			name, _ = meta.NewAccessor().GenerateName(obj)
+		}
+	}
+
+	items, err := l.GetLimitRanges(a)
+	if err != nil {
+		return err
+	}
+
+	// ensure it meets each prescribed min/max
+	for i := range items {
+		limitRange := items[i]
+
+		if !l.actions.SupportsLimit(limitRange) {
+			continue
+		}
+
+		err = l.actions.ValidateLimit(limitRange, a.GetResource().Resource, a.GetObject())
+		if err != nil {
+			return admission.NewForbidden(a, err)
+		}
+	}
+	return nil
```

Comment:
They are really the same with the exception `l.actions.ValidateLimit` vs.  `l.actions.MutateLimit`? Maybe make that explicit?

Reply:
Much better now.